habitat/core/vector_env.py

```diff
@@ -291,14 +302,15 @@ def step_at(self, index_env: int, action: int):
         self._is_waiting = False
         return results
 
-    def async_step(self, actions: List[int]) -> None:
+    def async_step(self, data: List[Dict[str, Any]]) -> None:
```

Comment:
Let's push the backwards compatibility stuff into here and make `async_step` and `step` have the same signature